server/grpc.go

```diff
@@ -101,35 +112,65 @@ func (ss *StreamServer) Run() {
 				metrics.DecConnection()
 			case payload := <-ss.payloads:
 				ss.clientManager.SendPayload(payload)
+			case re := <-ss.resfreshEvents:
+				ss.clientManager.DeleteEvents(re.client)
+				re.client.SetEvents(re.events)
+				ss.clientManager.AddClient(*re.client)
 			}
 		}
 	}()
 }
 
-func (ss *StreamServer) Events(request *proto.Request, es proto.StreamService_EventsServer) error {
-	ss.accessLogger.Info("gRPC",
-		zap.Array("request-events", zapcore.ArrayMarshalerFunc(func(enc zapcore.ArrayEncoder) error {
-			for _, e := range request.Events {
-				enc.AppendString(e.Type)
+func (ss *StreamServer) Events(es proto.StreamService_EventsServer) error {
+	client := manager.NewClient([]string{})
+	ss.newClients <- client
+	go func() {
+		for {
+			request, err := es.Recv()
+			if err == io.EOF {
+				<-es.Context().Done()
+				return
 			}
-			return nil
-		})),
-	)
-	if request == nil || request.Events == nil {
-		return errors.New("request can't be nil")
-	}
 
-	l := len(request.Events)
-	events := make([]string, l)
-	for i := 0; i < l; i++ {
-		events[i] = request.Events[i].Type
-	}
+			if err != nil {
+				if grpc.Code(err) != codes.Canceled {
+					ss.errChan <- errors.Wrap(err, "Recv error")
+					return
+				} else {
```

Comment:
__[golint]__
* [golint] if block ends with a return statement, so drop this else and outdent its block

<!-- Comment by [SideCI](https://sideci.com) -->